aws/lambda/log-classifier/src/bedrock.rs

```diff
@@ -0,0 +1,79 @@
+mod prompts;
+
+use aws_config::meta::region::RegionProviderChain;
+use aws_sdk_bedrockruntime::types::ConversationRole::User;
+use aws_sdk_bedrockruntime::types::ContentBlock;
+use aws_sdk_bedrockruntime::types::Message;
+use prompts::FIND_ERROR_LINE_PROMPT;
+use insta::assert_snapshot;
+use aws_sdk_bedrockruntime::operation::converse::ConverseOutput;
+use aws_sdk_bedrockruntime::operation::converse::ConverseError;
+use aws_config::BehaviorVersion;
+use aws_smithy_runtime_api;
+use aws_smithy_runtime_api::client::result::SdkError;
+use aws_smithy_runtime_api::http::Response;
+use tokio::fs;
+// Creates a snippet of that is n lines long. The end of the snippet is the specified error line.
+// Input: log_lines: Vec<&str>, error_line: &str, num_lines: usize
+// Output: Vec<&str>
+fn create_log_snippet<'a>(log_lines: Vec<&'a str>, error_line: &str, num_lines: usize) -> Vec<&'a str> {
+    let mut snippet: Vec<&str> = Vec::new();
+    
+    // Find the index of the error line
+    let error_line_index = log_lines.iter().position(|&line| line == error_line);
+    
+    // If error_line is not found, return an empty vector
+    if let Some(error_index) = error_line_index {
+        let start_index = error_index.saturating_sub(num_lines);
+        let end_index = (error_index + num_lines + 1).min(log_lines.len());
+        
+        snippet.extend(&log_lines[start_index..end_index]);
+    }
+    
+    snippet
+}
+
+async fn make_query(input_text: &String) -> Result<ConverseOutput, SdkError<ConverseError, Response>> {
+    let region_provider = RegionProviderChain::default_provider().or_else("us-west-2");
+    let config = aws_config::load_defaults(BehaviorVersion::v2024_03_28()).await;
+    let client = aws_sdk_bedrockruntime::Client::new(&config);
+    let prompt = FIND_ERROR_LINE_PROMPT.replace("{{LOG_SNIPPET}}", input_text);
```

Comment:
Just a thought that I found https://docs.rs/minijinja/latest/minijinja/ in case the prompt snippet becomes more complex